lldb/source/API/SBDebugger.cpp

```diff
@@ -1089,9 +1089,9 @@ void SBDebugger::SetSelectedTarget(SBTarget &sb_target) {
   Log *log = GetLog(LLDBLog::API);
 
   TargetSP target_sp(sb_target.GetSP());
-  if (m_opaque_sp) {
+  if (m_opaque_sp && target_sp)
     m_opaque_sp->GetTargetList().SetSelectedTarget(target_sp);
```

Comment:
Actually, is there any circumstance in which we'd want to set the selected target to an empty or invalid target?  I don't think so, in which case we should probably do that checking in TargetList::SetSelectedTarget.

Reply:
Had a discussion with Greg earlier:

> A default constructed SBTarget should have an invalid shared pointer, so this shouldn't ever try to be added to the list...

So I think the change here is good